backend/src/src-cvesearch/src/main/java/com/bosch/osmi/sw360/cvesearch/datasink/VulnerabilityConnector.java

```diff
@@ -0,0 +1,192 @@
+            dbVulnerability = vulnerability;
+        }
+        if(RequestStatus.FAILURE.equals(vulRequestStatus)){
+            return UpdateType.FAILED;
+        }
+        partiallyFilledRelation.setVulnerabilityId(dbVulnerability.getId());
+        RequestStatus requestStatus = addOrUpdate(partiallyFilledRelation);
+        if(RequestStatus.FAILURE.equals(requestStatus)){
+            return UpdateType.FAILED;
+        }
+        return updateType;
+    }
+
+    public VulnerabilityUpdateStatus addOrUpdate(Vulnerability vulnerability,
+                                                 ReleaseVulnerabilityRelation partiallyFilledRelation,
+                                                 VulnerabilityUpdateStatus currentVulnerabilityUpdateStatus){
+        UpdateType updateType = addOrUpdate(vulnerability, partiallyFilledRelation);
+        currentVulnerabilityUpdateStatus.getStatusToVulnerabilityIds()
+                .get(updateType)
+                .add(vulnerability.getExternalId());
+        return currentVulnerabilityUpdateStatus;
+    }
+
+    public Map<UpdateType, List<Vulnerability>> addOrUpdateVulnerabilityAndSetId(
+            Vulnerability vulnerability,
+            Map<UpdateType, List<Vulnerability>> statusToVulnerabilities){
+
+        Vulnerability dbVulnerability = vulnerabilityDatabaseHandler.getByExternalId(Vulnerability.class, vulnerability.getExternalId());
+        if(dbVulnerability != null){
+            if(! isMoreRecent(vulnerability, dbVulnerability)) {
+                statusToVulnerabilities.get(UpdateType.OLD).add(dbVulnerability);
+                return statusToVulnerabilities;
+            } else {
+                dbVulnerability = VulnerabilityMapper.updateFromVulnerability(dbVulnerability, vulnerability);
+                RequestStatus requestStatus = vulnerabilityDatabaseHandler.update(dbVulnerability);
+                if (RequestStatus.SUCCESS.equals(requestStatus)){
+                    statusToVulnerabilities.get(UpdateType.UPDATED).add(dbVulnerability);
+                    return statusToVulnerabilities;
+                }
+             }
+        } else {
+            RequestStatus requestStatus = vulnerabilityDatabaseHandler.add(vulnerability);
+            if (RequestStatus.SUCCESS.equals(requestStatus)){
+                statusToVulnerabilities.get(UpdateType.NEW).add(vulnerability);
+                return statusToVulnerabilities;
+            }
+        }
+        statusToVulnerabilities.get(UpdateType.FAILED).add(vulnerability);
+        return statusToVulnerabilities;
+    }
+
+
+    public Map<UpdateType, List<Vulnerability>> addOrUpdateVulnerabilitiesAndSetIds(List<Vulnerability> vulnerabilities) {
+        Map<UpdateType, List<Vulnerability>> statusToVulnerabilities = new HashMap<>();
+        for (UpdateType updateType: UpdateType.values()){
+            statusToVulnerabilities.put(updateType, new ArrayList<>());
```

Comment:
This same initialization of `Map<UpdateType, List<Vulnerability>>` and a very similar one of `Map<UpdateType, List<String>>` seems to be done multiple times in the code. There's some refactoring potential, I believe.